src/Unused/CLI/Util.hs

```diff
@@ -1,11 +1,28 @@
 module Unused.CLI.Util
     ( resetScreen
+    , withInterruptHandler
     , module System.Console.ANSI
     ) where
 
 import System.Console.ANSI
+import Control.Exception (throwTo)
+import System.Posix.Signals (Handler(Catch), installHandler, keyboardSignal)
+import Control.Concurrent (ThreadId, myThreadId)
+import System.Exit (ExitCode(ExitFailure))
 
 resetScreen :: IO ()
 resetScreen = do
     clearScreen
     setCursorPosition 0 0
+
+withInterruptHandler :: IO () -> IO ()
+withInterruptHandler body = do
+    tid <- myThreadId
+    _ <- installHandler keyboardSignal (Catch (handleInterrupt tid)) Nothing
+    body
+
+handleInterrupt :: ThreadId -> IO ()
+handleInterrupt tid = do
+    resetScreen
+    showCursor
+    throwTo tid $ ExitFailure 130
```

Comment:
Is `130` something expected by other software?

This is the list of exit codes that the BSDs (including OS X) care about: http://man.openbsd.org/sysexits

An interrupt is probably not covered in that list; I'd have gone with `1`. I tend to keep it under `128` because I don't trust every path the exit code travels. Here's an example thread where they discuss `system()` in Perl and how the numbers go through the maze of unix to get there: http://www.perlmonks.org/?node_id=822569